src/api/restApi/settings/settings.ts

```diff
@@ -13,27 +14,31 @@ export const getSettings = () => get<Settings>(`${endpoint}`, { ...defaultParams
 export const addToWhitelist = (body: WhitelistBody, signature: string) => post(`${endpoint}/allowed_list/${body.account}`, null, { headers: { ...defaultParams.headers, Authorization: `Bearer ${signature}` }, ...defaultParams });
 
 export const useSettings = () => {
-  const [settings, setSettings] = useState<Settings | undefined>();
   const [isFetching, setIsFetching] = useState<boolean | undefined>(false);
   const [fetchingError, setFetchingError] = useState<ResponseError | undefined>();
 
-  const fetchSettings = useCallback(async () => {
+  const fetchSettings = useCallback(async (blockchains: Record<string, Chain>) => {
     setIsFetching(true);
     try {
-      const response = await getSettings();
+      const settingsSet = await Promise.all(Object.values(blockchains).map(async (blockchain) => {
+        axios.defaults.baseURL = blockchain.apiEndpoint;
```

Comment:
let's remove it, might get us in some trouble with all async shenanigans 

const { currentSettings } = useSettings()
const { baseUrl } = currentSettings;
const offers = await getOffers(baseUrl);